### PR TITLE
feat: CO-17 - UI: Learn more about our credit card options

### PR DESCRIPTION
**Pull request by ghost**
## PR Area

- [x] Frontend
- [ ] Testing
- [ ] Other: 

## Type of change

_What kind of change does this PR introduce?_

- [ ] Bugfix
- [x] Feature
- [ ] Other: 

## Description
* Readme, add more details to folder structure
* Decouple context-specific styling from buttons to make them more reusable
* Rename IconCard to Card and put it in dataDisplay
* Adds `size` props to buttons and cards - auto resizes from sm-lg based on breakpoints, the other sizes are static
* Adds stories for Card and improved visibility for card and button sizes
* Add padding to dialogs for small screens
* Split cards styling into a separate scss for better reusability
* Adds row of credit card options in the dialog
* improved visibility of enums and separation from props by putting enums in `src/models/enums`
* Adds more todos

## Security & risks

- [ ] Adds new third party libraries
- [ ] Adds new `.env` variables
- [ ] Integrates with Third-party apis (dependency must be resilient)
- [ ] Has specific [OWASP security concerns](https://owasp.org/www-project-top-ten/) - list them: (also see [Fixing OWASP Top 10](https://nodegoat.herokuapp.com/tutorial))

## Pre-publish checklist

- [x] I have assigned myself to the ticket
- [x] I have performed a self-review of my code
- [ ] I have performed relevant browser testing
- [ ] I have performed keyboard testing of new or modified interactive components

#### Evidence screenshots
Screen Shot 2021-09-01 at 11 31 42 am
Screen Shot 2021-09-01 at 11 31 56 am
Screen Shot 2021-09-01 at 11 32 45 am
Screen Shot 2021-09-01 at 11 32 48 am
Screen Shot 2021-09-01 at 11 32 59 am
Screen Shot 2021-09-01 at 11 33 01 am

## Reviewer checklist

- [ ] I have run the code (if applicable)
- [ ] I have reviewed and run any tests (if applicable)

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -25,7 +29,11 @@ export default defineComponent({
   },
   props: {
     productPrice: Number,
-    lang: String,
+    lang: String as () => LanguageCodeEnum,
+    theme: {
+      type: String as () => ThemeEnum,
+      default: (props: AppProps) => props.lang,
```
Comment (ghost) [inline src/App.vue:35]: When testing different themes in development this default prop can be changed. e.g: 
```
default: (props: AppProps) => 'qmc',
```
or
```
default: (props: AppProps) => 'farmers',
```

File: src/components/buttons/Button.vue [hunk 2]
```diff
@@ -52,19 +62,38 @@ export default defineComponent({
   appearance: none;
   font-size: 13px;
   border-radius: var(--radius-1);
-  min-width: 138px;
-  min-height: 35px;
   justify-content: center;
-  margin-top: 5px;
 
-  @media (min-width: 430px) {
-    margin-top: 0;
```
Comment (ghost) [inline src/components/buttons/Button.vue:61]: Moving margin-top out of the styles here makes the button more reusable. The margin now comes from WidgetMainSmall

File: src/widgets/WidgetMainSmall.vue [hunk 3]
```diff
@@ -157,29 +209,19 @@ Indicative monthly payment is a minimum monthly repayment (MMP) of the greater o
...
-    display: flex;
-    position: absolute;
-    top: 10px;
-    left: 8px;
-  }
-
   &__close {
-    position: absolute;
-    top: 10px;
-    right: 10px;
```
Comment (ghost) [inline src/widgets/WidgetMainSmall.vue:179]: I changed all of these from absolute positioning to static, and solved it with flex, margin and padding instead. It's a bit more maintainable and easier to control spacing.

**Review by antoninroam — APPROVED** [review]

File: src/models/enums/SizeEnum.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+export enum SizeEnum {
+  Auto = 'auto',
+  Small = 'sm',
+  Medium = 'md',
+  Large = 'lg',
+}
+
+export type Sizes = `${SizeEnum}`
```
Comment (jpyepez-roam) [inline src/models/enums/SizeEnum.ts:8]: Nice!!

**Review by jpyepez-roam — APPROVED** [review]
Looks great! Approved 👍

**Review by arjun-at-roam — APPROVED** [review]
Verified locally. 🔥

**Pull request merged**